### PR TITLE
Add NonreflectableMessage and support Protobuf v5

### LINKED ISSUE #2757: Protobuf v5 support
**Is your feature request related to a problem? (你需要的功能是否与某个问题有关?)**

The brpc vcpkg port depends on protobuf. Since vcpkg maintains a single-version scheme, updating protobuf requires brpc to support it.

**Describe the solution you'd like (描述你期望的解决方法)**

Protobuf v5 support. V4 support can be broken in the process.

**Describe alternatives you've considered (描述你想到的折衷方案)**

Patching brpc, see https://github.com/microsoft/vcpkg/blob/052b06a95da4cee184bab4422e02e7f316055a99/ports/brpc/protobuf.patch

**Additional context/screenshots (更多上下文/截图)**

### PR DESCRIPTION
**Pull request by wasphin**
See https://github.com/apache/brpc/pull/2722#issuecomment-2272559689 inspired by unreflectable_message of @oathdruid.

### What problem does this PR solve?

Issue Number: #2757

Problem Summary:

隔离 Protobuf Message API，简化内部消息类型实现，简化后续 Protobuf 适配工作，支持 Protobuf 到 v5.28.0

### What is changed and the side effects?

Changed:

Side effects:
- Performance effects(性能影响):

- Breaking backward compatibility(向后兼容性): 

---
### Check List:
- Please make sure your changes are compilable(请确保你的更改可以通过编译).
- When providing us with a new feature, it is best to add related tests(如果你向我们增加一个新的功能, 请添加相关测试).
- Please follow [Contributor Covenant Code of Conduct](https://github.com/apache/brpc/blob/master/CODE_OF_CONDUCT.md).(请遵循贡献者准则).

### CODE REVIEW

File: src/brpc/esp_message.h [hunk 1]
```diff
@@ -18,64 +18,32 @@
...
-#include "butil/iobuf.h"       
-#include "brpc/proto_base.pb.h"
-#include "brpc/pb_compat.h"
+#include "brpc/message_helper.h"
+#include "butil/iobuf.h"
 
 namespace brpc {
 
-class EspMessage : public ::google::protobuf::Message {
+class EspMessage : public MessageHelper<EspMessage>::BaseType {
```
Comment (chenBright) [inline src/brpc/esp_message.h:27]: MessageHelper有什么作用呢？目前来说，是不是直接继承NonreflectableMessage<T>就可以了？

Comment (wasphin) [reply src/brpc/esp_message.h:27]: 开始是计划针对不同环境可以继承 Message，不过后来看是没什么必要，可以去掉

**Comment by wasphin** [comment]
ci 问题看和改动无关，暂时没有 bazel 测试环境，得晚几天再看看

**Comment by wwbmmm** [comment]
LGTM

**Comment by oathdruid** [comment]
> ci 问题看和改动无关，暂时没有 bazel 测试环境，得晚几天再看看

看起来boost.uuid@1.83.0还是有点问题，[PR](https://github.com/bazelbuild/bazel-central-registry/pull/2953)预期boost.uuid@1.83.0.bcr.1能修复；我先给这个发布到babylon的registry了，应该能先跑起来；等bcr的merge进去可以无缝替换

**Review by chenBright — APPROVED** [review]
LGTM

**Pull request merged**